.travis.yml

```diff
@@ -3,68 +3,25 @@ compiler:
   - clang
   - gcc
 env:
+  - TARGET=coverage
   - TARGET=debug
   - TARGET=debug.nounity
   # We can specify any combination of builds here, for example, to
   # include release builds, too, uncomment the following lines.
   #- TARGET=release
   #- TARGET=release.nounity
+matrix:
+  exclude:
+    # Because gcov won't work (easily) with clang
+    - env: TARGET=coverage
+      compiler: clang
+    # Because coverage target is basically debug + --coverage flags
+    - env: TARGET=debug
+      compiler: gcc
 before_install:
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq python-software-properties
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo add-apt-repository -y ppa:afrank/boost
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq g++-4.8
-  - sudo apt-get install -qq libboost1.57-all-dev
-  - sudo apt-get install -qq mlocate
-  - sudo updatedb
-  - sudo locate libboost | grep /lib | grep -e ".a$"
-  - sudo apt-get install -qq protobuf-compiler libprotobuf-dev libssl-dev exuberant-ctags
-  # We need gcc >= 4.8 for some c++11 features
-  - sudo apt-get install -qq gcc-4.8
-  - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 40 --slave /usr/bin/g++ g++ /usr/bin/g++-4.8
-  - sudo update-alternatives --set gcc /usr/bin/gcc-4.8
-  # Stuff is gold. Nuff said ;)
-  - sudo apt-get -y install binutils-gold
-  # We can get a backtrace if the guy crashes
-  - sudo apt-get -y install gdb
-  # What versions are we ACTUALLY running?
-  - g++ -v
-  - clang -v
-  # Avoid `spurious errors` caused by ~/.npm permission issues
-  # Does it already exist? Who owns? What permissions?
-  - ls -lah ~/.npm || mkdir ~/.npm
-  # Make sure we own it
-  - sudo chown -R $USER ~/.npm
-
+  - bin/ci/ubuntu/install-dependencies.sh
```

Comment:
Oh, I think we will be using it heaps, cause we can look at new code in pull wrecks and see if it's covered. 

I hope this is merged in ASAP because it will be useful for all subsequent pull requests.